src/containers/IntentService.jsx

```diff
@@ -156,36 +72,31 @@ class IntentService extends Component {
         )}
         <div className="coz-service-layout">
           <ServiceBar
-            appEditor={data.cozyAppEditor}
-            appName={data.cozyAppName}
-            iconPath={`../${data.cozyIconPath}`}
-            onCancel={() => this.cancel()}
+            appEditor={appData.cozyAppEditor}
+            appName={appData.cozyAppName}
+            iconPath={`../${appData.cozyIconPath}`}
+            onCancel={onCancel}
             closeable={closeable}
             hasReturnToKonnectorsListButton={
-              !isFetching && !error && konnectorsList.length > 1 && konnector
+              !error && konnectors.length > 1 && konnector
             }
             returnToKonnectorsList={() => this.setState({ konnector: null })}
             {...this.context}
           />
-          {!isFetching &&
-            !error &&
-            konnector && (
-              <CreateAccountService
-                alertDeleteSuccess={() => this.setState({ konnector: null })}
-                konnector={konnector}
-                onCancel={() => this.setState({ konnector: null })}
-                onSuccess={() => this.setState({ konnector: null })}
-                disableSuccessTimeout={disableSuccessTimeout}
-                closeModal={() => this.setState({ konnector: null })}
-                {...this.context}
-              />
-            )}
-          {!isFetching &&
-            !error &&
-            konnectorsList.length > 1 &&
-            !konnector && (
+          {konnector && (
+            <CreateAccountService
+              konnector={konnector}
+              onCancel={() => this.onCancel()}
+              onSuccess={account => this.onSuccess(account)}
+              disableSuccessTimeout={disableSuccessTimeout}
+              closeModal={() => this.onCancel()}
+              {...this.context}
+            />
+          )}
+          {!konnector &&
+            konnectors.length > 1 && (
```

Comment:
It looks like this component behaves very differently depending on the number of `konnectors` (eg. what it renders or the multiple `isUnique` checks). Have you considered splitting it into dumber components?

Reply:
Totally, but as it is not the true meaning of this PR, I limited myself to the minimum rewriting/refactoring, which is already big. But for sure we will improve this part in a near future.

Reply:
Fair enough, the refactoring has to stop somewhere :)